auth/cache.go

```diff
@@ -0,0 +1,59 @@
+package auth
+
+import (
+	"math/rand"
+	"sync"
+)
+
+// Cache represents a random replacement cache.
+type Cache struct {
+	size  int                 // Maximum size where this cache can potentially grow upto.
+	keys  []*string           // A slice of keys for choosing a random key for eviction.
```

Comment:
Is there a reason this is a pointer? I can't see anywhere it's used as such, it's never nil checked and always dereferenced.

Reply:
This doesn't need to be a pointer, since we're already keeping the set of keys in another map for fast lookup, I though a reference to that value would be ideal.